### ISSUE TITLE
Infinite recursion on nested #ifs

### ISSUE THREAD

**Issue by AndrewRayCode** [dy/prepr#5]
```javascript
    console.log(
      prepr(`
  #ifdef OUTER
    #ifdef INNER
      uniform float a;
    #else
      uniform float b;
    #endif
  #endif
`)
    );
```

>     RangeError: Maximum call stack size exceeded
>         at String.replace (<anonymous>)
> 
>       at process (node_modules/prepr/index.js:70:19)
>       at process (node_modules/prepr/index.js:124:22)
>       at process (node_modules/prepr/index.js:124:22)
>       at process (node_modules/prepr/index.js:124:22)
>       at process (node_modules/prepr/index.js:124:22)
>       at process (node_modules/prepr/index.js:124:22)
>       at process (node_modules/prepr/index.js:124:22)

**Comment by AndrewRayCode** [comment]
I think it's because the code that looks for the `else` branch doesn't consider nested ifs:

https://github.com/dy/prepr/blob/master/index.js#L273-L278

```javascript
var matchElse;
if (matchElse = /^\s*#else[^\n\r]*$/m.exec(body)) {
    elseBody = body.slice(matchElse.index + matchElse[0].length);
    body = body.slice(0, matchElse.index);
}
```

`body` here is
```
#if OUTER
   #if INNNER
      ...
    #else
      ...
    #endif
#endif
```
And it only looks for the else by regex

**Comment by dy** [comment]
Should be fixed in  3c0fcc4
